File: code/jvm/src/main/kotlin/pt/isel/daw/channels/http/model/channel/ChannelOutputModel.kt
package pt.isel.daw.channels.http.model.channel

import pt.isel.daw.channels.domain.channels.Type
import pt.isel.daw.channels.domain.user.UserInfo

data class ChannelOutputModel (
    val id: Int,
    val name: String,
    val owner: UserInfo,
    val type: Type,
    val members: List<UserInfo>,
    val bannedMembers: List<UserInfo>
)

File: code/js/src/services/channels/models/ChannelOutputModel.ts
import { Type } from '../../../domain/channels/Type';
import { UserInfo } from '../../../domain/users/UserInfo';

export type ChannelOutputModel = {
  id: number;
  name: string;
  owner: UserInfo;
  type: Type;
  members: Array<UserInfo>;
  bannedMembers: Array<UserInfo>;
};
